### PR TITLE
feat: adding expiration date update on existing gpkg request (MAPCO-2311)

### PR DESCRIPTION
**Pull request by ronenkapelian**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

on case of duplication:
in progress
completed with isCleaned=False

should update expirationDate of job

### CODE REVIEW

File: src/clients/jobManagerWrapper.ts [hunk 1]
```diff
@@ -169,6 +169,29 @@ export class JobManagerWrapper extends JobManagerClient {
...
+    const getJobUrl = `/jobs/${jobId}`;
+    const newExpirationDate = new Date();
+    newExpirationDate.setDate(newExpirationDate.getDate() + this.expirationDays);
+
+    const job = await this.get<JobResponse | undefined>(getJobUrl);
+    if (job !== undefined) {
+      const expirationDate = new Date(job.expirationDate as Date);
+      if (expirationDate < newExpirationDate) {
+        await this.updateJobExpirationDate(jobId);
+        this.logger.debug(`old expiration date was updated`);
```
Comment (asafMasa) [inline src/clients/jobManagerWrapper.ts]: put the log before the update await method, should be info, please add some data - Updating (jobId, oldExpirationDate and newExpirationDate)

File: src/clients/jobManagerWrapper.ts [hunk 2]
```diff
@@ -169,6 +169,29 @@ export class JobManagerWrapper extends JobManagerClient {
...
+      }
+    }
+  }
+  private async updateJobExpirationDate(jobId: string): Promise<void> {
```
Comment (asafMasa) [inline src/clients/jobManagerWrapper.ts]: add enter (line) between methods

File: src/clients/jobManagerWrapper.ts [hunk 3]
```diff
@@ -169,6 +169,29 @@ export class JobManagerWrapper extends JobManagerClient {
...
+  public async validateAndUpdateExpiration(jobId: string): Promise<void> {
+    const getJobUrl = `/jobs/${jobId}`;
+    const newExpirationDate = new Date();
+    newExpirationDate.setDate(newExpirationDate.getDate() + this.expirationDays);
+
+    const job = await this.get<JobResponse | undefined>(getJobUrl);
+    if (job !== undefined) {
+      const expirationDate = new Date(job.expirationDate as Date);
+      if (expirationDate < newExpirationDate) {
+        await this.updateJobExpirationDate(jobId);
```
Comment (asafMasa) [inline src/clients/jobManagerWrapper.ts]: pass newExpirationDate as a parameter to the updateJobExpirationDate method
or replace it with
await this.put(getJobUrl, {
      newExpirationDate,
    });
   // where - getJobUrl needs to be renamed to getOrUpdateURL

File: src/clients/jobManagerWrapper.ts [hunk 4]
```diff
@@ -169,6 +169,29 @@ export class JobManagerWrapper extends JobManagerClient {
...
 
+  public async validateAndUpdateExpiration(jobId: string): Promise<void> {
+    const getJobUrl = `/jobs/${jobId}`;
+    const newExpirationDate = new Date();
+    newExpirationDate.setDate(newExpirationDate.getDate() + this.expirationDays);
+
+    const job = await this.get<JobResponse | undefined>(getJobUrl);
+    if (job !== undefined) {
+      const expirationDate = new Date(job.expirationDate as Date);
+      if (expirationDate < newExpirationDate) {
```
Comment (asafMasa) [inline src/clients/jobManagerWrapper.ts]: add info log where newExpirationDate > expirationDate and you don't update the expiration time

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good Job 👍, please mind my remarks

File: src/clients/jobManagerWrapper.ts (hunk 1 above)
Comment (ronenkapelian) [reply src/clients/jobManagerWrapper.ts]: done

File: src/clients/jobManagerWrapper.ts (hunk 2 above)
Comment (ronenkapelian) [reply src/clients/jobManagerWrapper.ts]: done

File: src/clients/jobManagerWrapper.ts (hunk 4 above)
Comment (ronenkapelian) [reply src/clients/jobManagerWrapper.ts]: done

**Review by asafMasa — APPROVED** [review]

**Pull request merged**